docs/migration-service.md

```diff
@@ -122,96 +201,288 @@ flowchart TD
+    - Comparing `env::signer_account_pk()` with the public key associated with the participant (note: this is different from the TLS key currently stored as [`signer_pk`](https://github.com/near/mpc/blob/b5a9d1b2eef4de47d19b66cb25b577da2b897560/crates/contract/src/tee/tee_state.rs#L32) in TEEState)
+    - Including this public key in the TEE attestation
+
+### Backup Service Components
+
+Both soft launch and hard launch implementations share common core components, with hard launch adding TEE-specific features and automation.
+
+#### Common Components (Both Soft and Hard Launch)
+
+1. **mTLS Client**: Establishes authenticated connections to MPC nodes using P2P keys
+   - Performs mutual TLS handshake using keys registered in the contract
+   - Validates peer identity against expected public key from contract
+
+2. **Symmetric Encryption**: Uses operator-provided `MPC_BACKUP_ENCRYPTION_KEY_HEX` for additional encryption layer
+   - Operator manually provides the same key to both MPC node and backup service
+   - Adds second layer of encryption beyond mTLS transport security
+   - Extra protection if contract state becomes inconsistent or manipulated
+
+#### Soft Launch-Specific Components
+
+3. **Local Storage**: Saves encrypted keyshares to disk
+   - Persists backed-up keyshares to local filesystem
+   - Enables recovery after process restart
+
+> The soft launch implementation is intentionally simple:
+> - **No TEE**: Runs on operator's standard infrastructure without hardware attestation
+> - **No blockchain interaction**: Does not query or submit transactions to the contract
+> - **No automatic monitoring**: Operator manually triggers backup/restore operations via CLI commands
+> - **Disk-based storage**: Encrypted keyshares persisted to local filesystem
+> - **Operator provides all context**: Node URL, public keys, and encryption keys passed as CLI arguments
+
+#### Hard Launch-Specific Components
+
+3. **Contract Transaction Interface**: Signs and submits transactions automatically
+   - Calls `register_backup_service()` with attestation periodically
+   - Uses account private key generated in TEE
+
+4. **TEE Runtime**: TDX-enabled environment backed by [dstack](https://github.com/Dstack-TEE/dstack)
+   - Generates hardware attestations proving execution in genuine TEE
+   - Protects cryptographic keys in hardware-encrypted memory
+   - Runs continuously (24/7) to maintain keyshares in memory
+   - Keeps keyshares in memory only: does not persist to disk as encryption key would be lost on restart, and operator must not access it
+   - Must re-fetch keyshares from MPC nodes after restart or power loss
+
+5. **Blockchain Monitor**: Maintains current view of MPC contract state
+   - Embedded NEAR light client or RPC connection to track contract state
```

Comment:
Hmm, why did you include an RPC connection here? I thought those are not good enough from a security pov.